fontir/src/orchestration.rs

```diff
@@ -34,83 +32,306 @@ bitflags! {
     }
 }
 
-pub type ContextItem<T> = Arc<RwLock<Option<Arc<T>>>>;
+impl Default for Flags {
+    /// Match the way gftools configures fontmake by default
+    fn default() -> Self {
+        Flags::EMIT_IR | Flags::PREFER_SIMPLE_GLYPHS
+    }
+}
 
-/// Generates fn $getter_name(&self) -> Arc<$value_type>.
+pub type ContextVec<T> = Arc<RwLock<Vec<T>>>;
+
+/// Clones are cheap and reference the same wrapped item courtesy of Arc
 ///
-/// Assumes we are in an impl block for a Context and that
-/// self.$lock_name is a ContextItem<$ir_type>
+/// Courtesy of Arc this is Clone even if T isn't
+#[derive(Default)]
+pub struct ContextItem<I, T, Ir>
+where
+    I: Clone + Debug + PartialEq + Eq + Hash,
+    Ir: IrIo<I>,
+{
+    id: I,
+    acl: Arc<AccessControlList<I>>,
+    irio: Arc<Ir>,
```

Comment:
maybe worth a comment noting what `irio` does? I assume serialization?

Reply:
Hm, I think I will try to rename to better reflect purpose

Reply:
Hopefully better